### DISCUSSION TITLE
intellisense 找不到 c++20 module

### DISCUSSION THREAD

**Discussion by bakabaka9405** [xmake-io/xmake#2565] [Q&A] [answered]
编译通过 xmake test 中的 hello 项目，确认不是代码和编译器问题。

尝试了 intellisense 和 clangd 均无果，有没有一种可能，是 compile_commands.json 没有声明 modules 文件？

**Comment by waruqi** [comment, marked as answer]
目前还不支持，看下
https://github.com/xmake-io/xmake/issues/1167
https://github.com/xmake-io/xmake/issues/2114

**Comment by waruqi** [comment]
compile_commands 现在支持了 https://github.com/xmake-io/xmake/pull/2641

但是，即使 compile_commands 支持，我也不能保证 intellisense 支持，目前 vscode cpptools 还不支持带有 c++modules 的 compile_commands，只有 clangd 支持。

https://github.com/xmake-io/xmake/issues/1167
https://github.com/microsoft/vscode-cpptools/issues/6302#issuecomment-1215554637
